### PR TITLE
Added single pod deployment to osdctl

### PR DESCRIPTION
**Pull request by luis-falcon**
https://issues.redhat.com/browse/OSD-6776

This change was made in order to allow for more targeted node captures. instead of sniffing in all nodes all the time, we can select the specific node in which our workloads are running. which will allow for smaller pcaps easier to examine. Also, We can have a way for people who do not have cluster admins (not able to deploy daemonsets) ways to sniff at a node level through pods (this is specifically built for SBR).

### CODE REVIEW

File: cmd/network/packet-capture.go [hunk 1]
```diff
@@ -325,3 +366,140 @@ func copyFilesFromPacketCapturePods(o *packetCaptureOptions) error {
...
+	return desired, nil
+}
+
+// createPacketCapturePod creates the given Pod resource
+func createPacketCapturePod(o *packetCaptureOptions, capturePod *corev1.Pod) error {
+	if err := o.kubeCli.Create(context.TODO(), capturePod); err != nil {
+		return fmt.Errorf("failed to create Pod %s/%s: %v", capturePod.Namespace, capturePod.Name, err)
+	}
+	return nil
+}
```
Comment (dustman9000) [inline cmd/network/packet-capture.go:492]: nit: consistent line endings between functions

Comment (dustman9000) [reply cmd/network/packet-capture.go:492]: Can you also add a comment above this function?

File: cmd/network/packet-capture.go [hunk 2]
```diff
@@ -58,6 +59,7 @@ func newCmdPacketCapture(streams genericclioptions.IOStreams, flags *genericclio
 	packetCaptureCmd.Flags().StringVarP(&ops.namespace, "namespace", "n", packetCaptureNamespace, "Namespace to deploy Daemonset")
 	packetCaptureCmd.Flags().StringVarP(&ops.nodeLabelKey, "node-label-key", "", nodeLabelKey, "Node label key")
 	packetCaptureCmd.Flags().StringVarP(&ops.nodeLabelValue, "node-label-value", "", nodeLabelValue, "Node label value")
+	packetCaptureCmd.Flags().BoolVarP(&ops.singleNode, "single-node", "", singleNode, "Boolean value to deploy a single pod or a deamonset (default true)")
```
Comment (dustman9000) [inline cmd/network/packet-capture.go:62]: Should be default 'false', right?
```suggestion
	packetCaptureCmd.Flags().BoolVarP(&ops.singleNode, "single-node", "", singleNode, "Boolean value to deploy a single pod or a deamonset (default false)")
```

File: cmd/network/packet-capture.go [hunk 3]
```diff
@@ -121,6 +133,35 @@ func (o *packetCaptureOptions) run() error {
 	return nil
 }
 
+func (o *packetCaptureOptions) runPod() error {
+
```
Comment (dustman9000) [inline cmd/network/packet-capture.go:137]: delete unnecessary line ending
```suggestion
```

File: cmd/network/packet-capture.go [hunk 4]
```diff
@@ -325,3 +366,140 @@ func copyFilesFromPacketCapturePods(o *packetCaptureOptions) error {
 
 	return nil
 }
+
+// desiredPacketCapturePod returns the desired Pod read in from manifests
+func desiredPacketCapturePod(o *packetCaptureOptions, key types.NamespacedName) *corev1.Pod {
```
Comment (dustman9000) [inline cmd/network/packet-capture.go:371]: Do you mind removing some of the unnecessary line endings in this function and compacting it?

**Comment by dustman9000** [comment]
Looks good, just some minor formatting comments. It also looks like there's some linting errors that need to be fixed.

File: cmd/network/packet-capture.go (hunk 2 above)
Comment (luis-falcon) [reply cmd/network/packet-capture.go:61]: Made the change on the code

Comment (luis-falcon) [reply cmd/network/packet-capture.go:61]: made change on the code.

**Comment by luis-falcon** [comment]
closing and opening a new one

**Pull request closed without merging**